### PR TITLE
*: add a region flag to vprox and fix some internal network interface…

### PR DESCRIPTION
**Pull request by adityamaru**
… resolution

----

> [!IMPORTANT]
> Add `region` flag to `vprox` server, modify internal network interface resolution, and update iptables rules for `us-west` region.
> 
>   - **Behavior**:
>     - Add `--region` flag to `serverCmdArgs` in `cmd/server.go`, making it a required parameter.
>     - Remove AWS-specific IP polling logic from `cmd/server.go`.
>     - Modify `runServer()` to pass `region` to `NewServerManager()`.
>   - **Network Interface**:
>     - Add `getInternalInterface()` in `lib/iputils.go` to resolve internal network interfaces.
>     - Use `getInternalInterface()` in `Server.InitState()` in `lib/server.go` for `us-west` region.
>   - **Iptables**:
>     - Add SNAT and FORWARD iptables rules for internal traffic in `Server.StartIptables()` in `lib/server.go` for `us-west` region.
>     - Remove these rules in `Server.CleanupIptables()`.
>   - **Server Management**:
>     - Update `NewServerManager()` in `lib/server_manager.go` to accept `region` parameter.
>     - Pass `region` to `Server` instances in `ServerManager.Start()`.
>   - **CI/CD**:
>     - Add new GitHub Actions workflow `build.yaml` for building the project.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=useblacksmith%2Fvprox&utm_source=github&utm_medium=referral)<sup> for 0b08cb9927d6e35cad8b562f6ef39efa7297e764. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by aayushshah15 — COMMENTED** [review]
Anyway we could add a regression test for this?

**Comment by adityamaru** [comment]
> Anyway we could add a regression test for this?

going to be kinda hard, cause it relies on all sorts of network calls. regression test for what in particular?

**Comment by aayushshah15** [comment]
> regression test for what in particular

Was imagining something with heavy mocking, but maybe that's not very useful

File: lib/server.go [hunk 1]
```diff
@@ -332,43 +345,45 @@ func (srv *Server) StartIptables() error {
...
-	}
-	if err := srv.Ipt.AppendUnique("nat", "POSTROUTING", rule...); err != nil {
-		return fmt.Errorf("failed to add SNAT rule: %v", err)
-	}
+	// SNAT rule for internal network traffic. This is currently only applicable for boxes in
+	// the US.
+	if srv.Region == "us-west" {
+		rule = []string{
+			"-s", srv.WgCidr.String(),
+			"-d", "10.0.0.0/24",
```
[the remark this answers is not in the document]
Comment (aayushshah15) [reply lib/server.go]: This is a reasonable comment

[the remark this answers is not in the document]
Comment (adityamaru) [reply lib/server.go]: yeah was on my mind, let me plumb something through

**Pull request merged**